### PR TITLE
Add inplace versions for ge, gt, lt, le

### PR DESCRIPTION
**Pull request by vishwakftw**

### CODE REVIEW

**Comment by dlibenzi** [comment]
Adding a few tests for those will help make sure we get the semantics right ;)

**Comment by vishwakftw** [comment]
It seems that if the comparison operations are performed in-place, the `ScalarType` of the tensor doesn't change.

**Comment by vishwakftw** [comment]
This is a reproducing example obtained using LibTorch:

```
Input: 
-0.1591  0.7471
 0.5057  0.0069
[ Variable[CPUFloatType]{2,2} ]
Other: 
-0.1742  0.1020
 1.6900 -1.9262
[ Variable[CPUFloatType]{2,2} ]
Out-place result: 
 0  0
 0  0
[ Variable[CPUByteType]{2,2} ]
In-place result: 
 0  0
 0  0
[ Variable[CPUFloatType]{2,2} ]
```

The script used is this:
```c++

#include <torch/torch.h>
#include <iostream>

int main() {
  torch::Tensor self = torch::randn({2, 2});
  torch::Tensor other = torch::randn({2, 2});
  torch::Tensor cmp = self.eq(other);
  std::cout << "Input: \n" << self << std::endl;
  std::cout << "Other: \n" << other << std::endl;
  std::cout << "Out-place result: \n" << cmp << std::endl;
  self.eq_(other);
  std::cout << "In-place result: \n" << self << std::endl;
return 0;
}
```

**Comment by dlibenzi** [comment]
> It seems that if the comparison operations are performed in-place, the `ScalarType` of the tensor doesn't change.

Then you need to wrap the IR Node returned by the comparison op, with a Cast op back into original type.
The result of ComparisonOp is predicate.

**Comment by dlibenzi** [comment]
> This is a reproducing example obtained using LibTorch:
> 
> ```
> Input: 
> -0.1591  0.7471
>  0.5057  0.0069
> [ Variable[CPUFloatType]{2,2} ]
> Other: 
> -0.1742  0.1020
>  1.6900 -1.9262
> [ Variable[CPUFloatType]{2,2} ]
> Out-place result: 
>  0  0
>  0  0
> [ Variable[CPUByteType]{2,2} ]
> In-place result: 
>  0  0
>  0  0
> [ Variable[CPUFloatType]{2,2} ]
> ```
> The script used is this:
> 
> ```c++
> #include <torch/torch.h>
> #include <iostream>
> 
> int main() {
>   torch::Tensor self = torch::randn({2, 2});
>   torch::Tensor other = torch::randn({2, 2});
>   torch::Tensor cmp = self.eq(other);
>   std::cout << "Input: \n" << self << std::endl;
>   std::cout << "Other: \n" << other << std::endl;
>   std::cout << "Out-place result: \n" << cmp << std::endl;
>   self.eq_(other);
>   std::cout << "In-place result: \n" << self << std::endl;
> return 0;
> }
> ```

Can you test with a result where there are some 'true' values?

**Comment by vishwakftw** [comment]
> Can you test with a result where there are some 'true' values?
Sure, here is one:
```
Input: 
 0.5837  0.1879
-0.2525 -1.1518
[ Variable[CPUFloatType]{2,2} ]
Other: 
 0.5837  0.1879
-0.2525 -1.1518
[ Variable[CPUFloatType]{2,2} ]
Out-place result: 
 1  1
 1  1
[ Variable[CPUByteType]{2,2} ]
In-place result: 
 1  1
 1  1
[ Variable[CPUFloatType]{2,2} ]
```

I did `other = self.clone()`, instead of randomly creating them.

File: test/cpp/test_aten_xla_tensor.cpp [hunk 1]
```diff
@@ -264,6 +264,19 @@ TEST_F(AtenXlaTensorTest, TestGe) {
   });
 }
 
+TEST_F(AtenXlaTensorTest, TestGeInplace) {
+  at::Tensor a = GetTestTensor({2, 3});
+  at::Tensor b = a.clone();
```
Comment (dlibenzi) [inline test/cpp/test_aten_xla_tensor.cpp]: Do not simply clone 'a'.
I'd clone, then in one element I'd do -=1.0 and in another one a +=1.0, just to test all combinations.

Comment (vishwakftw) [reply test/cpp/test_aten_xla_tensor.cpp]: Oh sorry. I meant to do this for the copy of `a` only. Will initialize using `GetTestTensor`. Thank you for the comment.

**Comment by vishwakftw** [comment]
@dlibenzi hope the tests are fine now. Thank you for the comments thus far.

**Pull request merged**